target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -425,52 +449,45 @@ abstract contract Pyth is
         returns (PythStructs.PriceFeed[] memory priceFeeds)
     {
         unchecked {
-            {
-                uint requiredFee = getUpdateFee(updateData);
-                if (msg.value < requiredFee)
-                    revert PythErrors.InsufficientFee();
-            }
-
+            uint8 totalNumUpdates = 0;
             priceFeeds = new PythStructs.PriceFeed[](priceIds.length);
             for (uint i = 0; i < updateData.length; i++) {
                 if (
                     updateData[i].length > 4 &&
                     UnsafeBytesLib.toUint32(updateData[i], 0) ==
                     ACCUMULATOR_MAGIC
                 ) {
-                    bytes memory accumulatorUpdate = updateData[i];
                     uint offset;
                     {
                         UpdateType updateType;
                         (
                             offset,
                             updateType
                         ) = extractUpdateTypeFromAccumulatorHeader(
-                            accumulatorUpdate
+                            updateData[i]
                         );
 
                         if (updateType != UpdateType.WormholeMerkle) {
                             revert PythErrors.InvalidUpdateData();
                         }
                     }
+
                     bytes20 digest;
                     uint8 numUpdates;
-                    bytes memory encoded = UnsafeBytesLib.slice(
-                        accumulatorUpdate,
-                        offset,
-                        accumulatorUpdate.length - offset
-                    );
-
+                    bytes memory encoded;
                     (
                         offset,
                         digest,
-                        numUpdates
-                    ) = extractWormholeMerkleHeaderDigestAndNumUpdates(encoded);
+                        numUpdates,
+                        encoded
```

Comment:
I don't like that this method returns `encoded` now -- it makes the method signature for this function extremely confusing.

BTW I find the overall handling of byte arrays in this code to be confusing. You're using two different techniques to slice out bytes (creating new arrays pointing to slices + offset indexes), so it's very hard for the reader to understand what an index into an array represents.

Imo the best way to handle this would be to have 1 byte array (the whole updateData) and 1 offset that increments as bytes are consumed.

Reply:
(please don't address the comment about overall handling of byte arrays in this PR. You can do that separately if you'd like)

Reply:
changing this to return `encoded` was a workaround for hitting the stack too deep error. 

I spoke to @ali-bahjati regarding slicing out bytes vs just reading offsets and he said  the implementation being used should be dependent on whether or not we're "throwing away" the data or need to possibly re-read it. I agree that there's areas where the implementation could be more consistent but there's also a few bytesize & gas cost considerations I'm also taking into account. I'll revisit this in a separate PR